packages/truffle-contract/test/abiV2.js

```diff
@@ -0,0 +1,74 @@
+var assert = require("chai").assert;
+var util = require("./util");
+
+// attribution where it's due - parts of this test are from this gist:
+// https://gist.github.com/ricmoo/e38d4d71dff7156033922d2e5de88d37
+
+describe("ABIEncoderV2", function() {
```

Comment:
If you want to see how ABIEncoderV2 support is meant to work from the user's perspective, this is the place to look.